lib/services/access_request.go

```diff
@@ -1153,6 +1181,21 @@ func (m *RequestValidator) Validate(ctx context.Context, req types.AccessRequest
 		}
 	}
 
+	// verify that all kube resource requests meet kube request modes.
+	if len(req.GetRequestedResourceIDs()) > 0 {
+		var err error
+		rolesToRequest := req.GetRoles()
+		if len(rolesToRequest) == 0 {
+			rolesToRequest, err = m.applicableSearchAsRoles()
```

Comment:
won't applicableSearchAsRoles return too many roles here? if len(rolesToRequest) is 0, the roles to be requested will be determined below inside the `if m.opts.expandVars` block. We might need to filter out some roles instead of validating them.

Reply:
thanks, i misunderstood how this worked. i removed this part